WordPress/Sniffs/NamingConventions/ValidFunctionNameSniff.php

```diff
@@ -10,136 +10,134 @@
-     * @return void
-     */
-    protected function processTokenWithinScope(PHP_CodeSniffer_File $phpcsFile, $stackPtr, $currScope)
-    {
-        $className  = $phpcsFile->getDeclarationName($currScope);
-        $methodName = $phpcsFile->getDeclarationName($stackPtr);
-
-        // Is this a magic method. IE. is prefixed with "__".
-        if (preg_match('|^__|', $methodName) !== 0) {
-            $magicPart = substr($methodName, 2);
-            if (in_array($magicPart, $this->_magicMethods) === false) {
-                 $error = "Method name \"$className::$methodName\" is invalid; only PHP magic methods should be prefixed with a double underscore";
-                 $phpcsFile->addError($error, $stackPtr, 'MethodDoubleUnderscore');
-            }
-
-            return;
-        }
-
-        // PHP4 constructors are allowed to break our rules.
-        if ($methodName === $className) {
-            return;
-        }
-
-        // PHP4 destructors are allowed to break our rules.
-        if ($methodName === '_'.$className) {
-            return;
-        }
-
-        // If this is a child class, it may have to use camelCase.
-        if ( $phpcsFile->findExtendedClassName( $currScope ) || $this->findImplementedInterfaceName( $currScope, $phpcsFile ) ) {
-            return;
-        }
-
-        $methodProps    = $phpcsFile->getMethodProperties($stackPtr);
-        $scope          = $methodProps['scope'];
-        $scopeSpecified = $methodProps['scope_specified'];
-
-        if ($methodProps['scope'] === 'private')
-            $isPublic = false;
-        else
-            $isPublic = true;
-
-        // If the scope was specified on the method, then the method must be
-        // camel caps and an underscore should be checked for. If it wasn't
-        // specified, treat it like a public method and remove the underscore
-        // prefix if there is one because we can't determine if it is private or
-        // public.
-        $testMethodName = $methodName;
-        if ($scopeSpecified === false && $methodName{0} === '_') {
-            $testMethodName = substr($methodName, 1);
-        }
-
-        if (strtolower($testMethodName) !== $testMethodName) {
-            $suggested = preg_replace('/([A-Z])/', '_$1', $methodName);
-            $suggested = strtolower($suggested);
-            $suggested = str_replace('__', '_', $suggested);
-
-            $error = "Function name \"$methodName\" is in camel caps format, try '".$suggested."'";
-            $phpcsFile->addError($error, $stackPtr, 'FunctionNameInvalid');
-        }
-
-    }//end processTokenWithinScope()
+class WordPress_Sniffs_NamingConventions_ValidFunctionNameSniff extends PEAR_Sniffs_NamingConventions_ValidFunctionNameSniff {
+
+	private $_magicMethods = array(
+		'construct',
+		'destruct',
+		'call',
+		'callStatic',
+		'get',
+		'set',
+		'isset',
+		'unset',
+		'sleep',
+		'wakeup',
+		'toString',
+		'set_state',
+		'clone',
+		'invoke',
+		'debugInfo',
+	);
+
+
+	/**
```

Comment:
> One never needs more than a single blank line.

As a dyslectic, I prefer more space anytime which is why I'm quite fond of the WPCS as it's generally speaking quite spacious compared to some.
I always found it weird and kind of inconsistent that the WPCS "more space" philosophy did not go so far as to have the `else` on a new line by itself and two lines between functions.

If given the choice, I will personally always use two lines between functions. For me, it really increases the scan-ability - one line break = within a function, two line break = hang on! new function coming up.

The `Squiz.WhiteSpace.FunctionSpacing` sniff looks for number of lines before and after each function and can be tweaked to look for 1 line rather than 2, so would be a good candidate for potentially adding to `extra` (and I'll just happily turn it off again for any project I work on).

This rule does conflict with the `WordPress.WhiteSpace.ControlStructureSpacing.BlankLineAfterEnd` rule for the last function in a class if the function does not have a `} // end myfunction()` comment.
Still - the fact that it throws a message `Blank line found after control structure` when finding a blank line after the last method in a class might actually be a bug....